core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -235,6 +235,23 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
           }
 
           tryBuilder.map(_.result())
+        case o: ConfigObject =>
```

Comment:
`traverseTry` is a nice idea considering that we use it all over the place. @froesef what do you think of it? It's similar to a common pattern used in functional programming. The only think is that I prefer to not catch `Throwable` and I would switch that to `Exception` at least.